composer.json

```diff
@@ -64,5 +72,20 @@
     }
   ],
   "minimum-stability": "dev",
-  "prefer-stable": true
+  "prefer-stable": true,
+  "scripts": {
+    "post-install-cmd": [
+      "@prefix-dependencies"
+    ],
+    "post-update-cmd": [
+      "@prefix-dependencies"
+    ],
+    "prefix-dependencies": [
+      "php-scoper add-prefix --output-dir=./third-party --force --quiet",
+      "echo '{ \"autoload\": { \"classmap\": [\"\"] } }' > ./third-party/composer.json",
+      "@composer dump-autoload --working-dir ./third-party --no-dev --classmap-authoritative",
+      "echo '{ \"autoload\": { \"classmap\": [\"\"] } }' > ./includes/composer.json",
+      "@composer dump-autoload --working-dir ./includes --no-dev --classmap-authoritative"
```

Comment:
In Site Kit we only rely on the actual classmap, not the full Composer autoloader. Which is certainly less elegant and a more manual approach, but there's still some chance for conflicts when relying on the Composer autoloader, e.g. due to its `Composer\Autoload\ClassLoader` class which could exist in multiple projects in different versions.

Reply:
Interesting, just stumbled upon https://github.com/composer/composer/issues/3852 that discusses this.

https://github.com/dangoodman/composer-for-wordpress is mentioned there as an alternative to rolling one's own autoloader. I'm gonna give it a try.

Reply:
@schlessera Also curious to hear your thoughts/ideas on this

Reply:
Went with a simple solution to just prefix the `ClassLoader` namespace. Way easier than all the other methods.